### PR TITLE
Stabilize uavcan.node.port.List

### LINKED ISSUE #134: Stabilize v0.x data types in the `uavcan` namespace

### PR DESCRIPTION
**Pull request by pavel-kirienko**
Closes #134; the other v0 types can be stabilized after the release.

### CODE REVIEW

**Comment by pavel-kirienko** [comment]
@thirtytwobits nudge

**Comment by thirtytwobits** [comment]
Sorry... 👀

File: uavcan/node/port/SubjectIDList.1.0.dsdl [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+uint16 CAPACITY = SubjectID.1.0.MAX + 1
+
+bool[CAPACITY] mask
+# The index represents the identifier value. True -- present/used. False -- absent/unused.
+
+SubjectID.1.0[<256] sparse_list
+# A list of identifiers that can be used instead of the mask if most of the identifiers are unused.
+
+uavcan.primitive.Empty.1.0 total
```
Comment (thirtytwobits) [inline uavcan/node/port/SubjectIDList.1.0.dsdl:16]: Wait...how am I supposed t use this?

File: uavcan/node/port/SubjectIDList.1.0.dsdl [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# A list of subject identifiers.
+# The range of subject-ID is large, so using a fixed-size bitmask would make this type difficult to handle on
+# resource-constrained systems. To address that, we provide two extra options: a simple variable-length list,
+# and a special case that indicates that every subject-ID is in use.
+
+@union
+
+uint16 CAPACITY = SubjectID.1.0.MAX + 1
+
+bool[CAPACITY] mask
```
Comment (thirtytwobits) [inline uavcan/node/port/SubjectIDList.1.0.dsdl:10]: If I use the sparse_list am I required to also set the mask? There's a bit of ambiguity here on the requirements for using this type.

**Review by thirtytwobits — CHANGES_REQUESTED** [review]

**Comment by pavel-kirienko** [comment]
It's a union, only one field is active at any moment. The empty field is
just a tag.

**Comment by thirtytwobits** [comment]
> It's a union, only one field is active at any moment. The empty field is just a tag.

I'm an idiot. Sorry.

**Review by thirtytwobits — APPROVED** [review]

**Comment by Alcalalandscapingpavers** [comment]
#140 landscapingpaversalcala.wordpress.com

**Pull request merged**